packages/cojson/src/coValueCore/coValueCore.ts

```diff
@@ -915,15 +979,46 @@ export class CoValueCore {
   }
 
   getCurrentBranchName() {
-    return this.verified?.header.meta?.branch as string | undefined;
+    return this.verified?.branchName;
   }
 
   getCurrentBranchSourceId() {
-    return this.verified?.header.meta?.source as RawCoID | undefined;
+    return this.verified?.branchSourceId;
   }
 
   isBranch() {
-    return Boolean(this.getCurrentBranchSourceId());
+    return Boolean(this.verified?.branchSourceId);
+  }
+
+  hasBranch(name: string, ownerId?: RawCoID) {
+    // This function requires the meta information to be parsed, which might not be the case
+    // if the value content hasn't been loaded yet
+    this.parseNewTransactions(false);
+
+    const currentOwnerId = getBranchOwnerId(this);
+    return this.branches.some((item) => {
```

Comment:
The branch commits are optimized to contain only the information for future loadings.

For this reason they require multiple checks to fill the meaning of undefined values